### ISSUE TITLE
🐛 [FIREBASE_AUTH] PhoneAuth stoped working on latest release

### ISSUE THREAD

**Issue by diegogarciar** [FirebaseExtended/flutterfire#5941]
## Bug report
The function VerifyPhoneNumber is failing as it is expecting a value not present in the response map on iOS

<img width="1360" alt="Screen Shot 2021-04-26 at 2 09 21 p m" src="https://user-images.githubusercontent.com/17276155/116144072-1eb02700-a699-11eb-9217-197c1a9fa7d9.png">

**Describe the bug**

### Steps to reproduce

1. Call FirebaseAuth.instance.verifyPhoneNumber(....
2. See error message

### Expected behavior

The codeSent callback should be called

---

## Additional context
This was discovered just after updating my dependencies today

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale en-MX)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.55.2)
[✓] Connected device (2 available)
    ! Error: Errors were encountered while preparing your device for development. Please check the Devices and Simulators Window.  (code -1)

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
...
- firebase 9.0.1 [http http_parser js]
- firebase_analytics_platform_interface 2.0.0 [flutter meta]
- firebase_analytics_web 0.3.0 [firebase firebase_analytics_platform_interface flutter flutter_web_plugins meta]
- firebase_auth_platform_interface 4.2.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 1.1.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.0.0 [flutter meta plugin_platform_interface]
- firebase_core_web 1.0.2 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_crashlytics_platform_interface 3.0.2 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_platform_interface 2.1.3 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 1.0.6 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- firebase_storage_platform_interface 2.0.3 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 1.0.5 [async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
...
```

</details>

---

**Comment by diegogarciar** [comment]
It seems that making `forceResendingToken` optional fixes this as `codeSent` also expects it as optional but I don't know if this is the expected behavior.

`final int? forceResendingToken = arguments['forceResendingToken'];`

**Comment by diegogarciar** [comment]
I'm still testing but it also seems that authentication state is not persisting across hot restarts

**Comment by diegogarciar** [comment]
@Salakar 
Heres what I found
`FirebaseAuth.instance.authStateChanges()` is always called just once with unauthenticated state, not calling again even after logging in.

If I call `FirebaseAuth.instance.signOut()` when the app starts, authStateChanges events are received as normal after logging in.

**Comment by Peng-Qian** [comment]
same here

**Comment by dfdgsdfg** [comment]
https://github.com/FirebaseExtended/flutterfire/pull/5944

**Comment by markusaksli-nc** [comment]
Isn't the root cause for this https://github.com/FirebaseExtended/flutterfire/issues/4542?

**Comment by diegogarciar** [comment]
@Peng-Qian did you also experienced the issue with authStateChanges()?

@markusaksli-nc theres a second bug that I mention in the comments, should I raise a new issue for that or is it ok here?

**Comment by markusaksli-nc** [comment]
I'm seeing that `forceResendingToken` is still null but not seeing the issues you mentioned with `firebase_auth: ^1.1.2`. 

As for the stream subscription it does seem to be null across a hot restart. That should probably be a separate issue. `userChanges()` still seems to be working fine.

**Comment by markusaksli-nc** [comment]
Actually there is already an open issue for that https://github.com/FirebaseExtended/flutterfire/issues/5952

**Comment by Peng-Qian** [comment]
Hi @diegogarciar ,

I only experience the wrong type bug and i use following old version to avoid this bug

```dart
  firebase_core: 1.0.4
  firebase_auth: 1.1.1
```

I also notice there is unsolved bug when we use non-null-safety dart SDK #5464

**Comment by markusaksli-nc** [comment]
As I thought the official example doesn't reproduce this because it hasn't been migrated yet.

https://github.com/FirebaseExtended/flutterfire/blob/dc8f49fc60eef1ebddf40e41b72749a421daf915/packages/firebase_auth/firebase_auth/example/lib/main.dart#L5

After automigrating to nullsafety the issue is reproducible with 
```yaml
environment:
  sdk: '>=2.12.0 <3.0.0'

dependencies:
  flutter:
    sdk: flutter
  firebase_auth: ^1.1.2
```
<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.3.0-1.0.pre.168, on macOS 11.2.3 20D91
    darwin-arm, locale en-GB)
    • Flutter version 2.3.0-1.0.pre.168 at
      /Users/nevercode/development/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 93242d7548 (11 hours ago), 2021-04-28 19:19:05 -0700
    • Engine revision e7939e091e
    • Dart version 2.14.0 (build 2.14.0-48.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/nevercode/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/nevercode/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/nevercode/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (3 available)
    • Nevercode’s iPhone (mobile) • b668e524315069f3db3661ac11ff1f66afafebdb •
      ios            • iOS 14.5
    • macOS (desktop)             • macos                                    •
      darwin-arm64   • macOS 11.2.3 20D91 darwin-arm
    • Chrome (web)                • chrome                                   •
      web-javascript • Google Chrome 90.0.4430.93

• No issues found!
```
</details>

**Comment by kuromukira** [comment]
I have also experienced this bug

**Comment by nzackoya** [comment]
This is unacceptable! Is there anyway to fix this?

**Comment by mrgnhnt96** [comment]
Just updated my dependencies and came across this as well. Any news on when the hot fix is going to be released?

**Comment by misterdev** [comment]
A fix seems to have been merged, see https://github.com/FirebaseExtended/flutterfire/pull/5987

**Comment by nzackoya** [comment]
> A fix seems to have been merged, see #5987

Doesn't fix the issue

**Comment by nzackoya** [comment]
If anyone needs the #5944  hot fix NOW you can do it like this

`dependency_overrides:
  firebase_auth_platform_interface:
    git:
      url: https://github.com/JYeop/flutterfire.git
      path: packages/firebase_auth/firebase_auth_platform_interface`

**Comment by Oreldad** [comment]
Any news? 🙏

**Comment by finroly** [comment]
I am facing same issue on iOS. This is really a drawback of flutter where we can't do anything and just wait for it to be fixed.

**Comment by erf** [comment]
i believe its fixed now in `firebase_auth: ^1.1.3`

**Comment by Salakar** [comment]
This has been fixed on the latest release,  thanks
